corehq/apps/data_dictionary/models.py

```diff
@@ -51,6 +51,21 @@ class CasePropertyGroup(models.Model):
     description = models.TextField(default='', blank=True)
     index = models.IntegerField(default=0, blank=True)
 
+    @classmethod
+    def get_or_create(cls, name, domain, case_type):
+        key = 'data-dict-group-{domain}-{type}-{name}'.format(
+            domain=domain, type=case_type, name=name
+        )
+        with CriticalSection([key]):
+            try:
+                group_obj = CasePropertyGroup.objects.get(
```

Comment:
I agree. I think it is unnecessary to use `CriticalSection`; `get_or_create` should do what you want.

Reply:
Offlined with @mkangia, I don't think that custom `get_or_create` would be required for this because django already provides this method.

Reply:
This change removes the usage of my custom get_or_create method. Also this avoids fetching case_type since it is already present on case property. [c4f4de2](https://github.com/dimagi/commcare-hq/pull/32799/commits/c4f4de2a5cd107415a920ea3f14260830163798a)